src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/FindAllReferences/FindAllReferencesHelper.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.Razor.DocumentMapping;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Razor.FindAllReferences;
+
+internal static class FindAllReferencesHelper
+{
+    public static async Task<string?> GetResultTextAsync(IDocumentMappingService documentMappingService, ISolutionQueryOperations solutionQueryOperations, int lineNumber, string filePath, CancellationToken cancellationToken)
+    {
+        // Roslyn will have sent us back text that comes from the .g.cs file, but that is often not helpful. For example give:
+        //
+        // <SurveyPrompt Title="Blah" />
+        //
+        // A FAR for the Title property will return just the word "Title" in the Text of the reference item, which does not
+        // help the user reason about the result. For such cases, its better to return the text from the Razor file, even
```

Comment:
Where does classification happen in FAR? Can we provide the classification?

Reply:
Was just logging an issue for that :)

Reply:
> https://github.com/dotnet/razor/issues/11286